firebase-storage/src/main/java/com/google/firebase/storage/StorageReference.java

```diff
@@ -563,14 +566,124 @@ public StreamDownloadTask getStream(@NonNull StreamDownloadTask.StreamProcessor
   /**
    * Deletes the object at this {@link StorageReference}.
    *
-   * @return a {@link Task} that indicates whether the operation succeeded or failed.
+   * @return A {@link Task} that indicates whether the operation succeeded or failed.
    */
+  @NonNull
   @PublicApi
   public Task<Void> delete() {
     TaskCompletionSource<Void> pendingResult = new TaskCompletionSource<>();
     StorageTaskScheduler.getInstance().scheduleCommand(new DeleteStorageTask(this, pendingResult));
     return pendingResult.getTask();
   }
+
+  // region List
+
+  /**
+   * List up to `maxResults` items (files) and prefixes (folders) under this StorageReference.
+   *
+   * <p>"/" is treated as a path delimiter. Firebase Storage does not support invalid object paths
+   * that end with "/" or contain two consecutive "/"s. All invalid objects in GCS will be filtered.
```

Comment:
One more thing.
Can we refer to them as `unsupported objects` to stay consistent?

Reply:
Done